File: 基础物理实验/粘滞系数2/calc.m
% 风速-频率数据
v = [0.51 0.66 1.10 1.47 1.76 2.02 2.24 2.48 2.71 2.87 3.06 3.28]; % m/s
f = [7.85 10.58 15.63 20.05 24.04 27.62 30.86 33.94 36.76 39.47 41.67 43.86]; % Hz

% 线性拟合（v = k*f + b）
p = polyfit(f, v, 1);  % 注意这里自变量是 f
k = p(1);  % 斜率
b = p(2);  % 截距

% 生成拟合曲线
f_fit = linspace(min(f), max(f), 100);
v_fit = polyval(p, f_fit);

% 绘图
figure;
plot(f, v, 'o', 'MarkerFaceColor', 'b', 'DisplayName', '实验数据');
hold on;
plot(f_fit, v_fit, 'r-', 'LineWidth', 1.5, ...
    'DisplayName', sprintf('拟合: v = %.4f f + %.4f', k, b));
xlabel('频率 f (Hz)');
ylabel('风速 v (m/s)');
title('风速-频率 线性拟合（v 对 f）');
legend('Location', 'best');
grid on;

% 输出拟合参数
fprintf('线性拟合方程: v = %.4f * f + %.4f\n', k, b);

File: 基础物理实验/霍尔效应/calc.py
import numpy as np
import matplotlib.pyplot as plt

# ================= 中文与负号 =================
plt.rcParams['font.sans-serif'] = ['SimHei']
plt.rcParams['axes.unicode_minus'] = False
# =============================================

# 实验数据（来自表格最后两列）
B = np.array([52.1, 102.05, 152.7, 202.6, 253.925, 305.0, 365.525, 417.55])  # mT
U_H = np.array([-11.725, -23.35, -34.65, -46.075, -57.9, -70.025, -83.75, -95.875])  # mV

# ================= 最小二乘线性拟合 =================
k, b = np.polyfit(B, U_H, 1)      # 斜率和截距
r = np.corrcoef(B, U_H)[0, 1]     # 相关系数

# 拟合曲线
B_fit = np.linspace(B.min(), B.max(), 200)
U_fit = k * B_fit + b

# ======================== 作图 ========================
plt.figure(figsize=(6.5, 4.2))

# 实验数据点
plt.scatter(B, U_H, color='r', s=35, label='data')

# 拟合直线（细线）
plt.plot(
    B_fit, U_fit,
    color='b',
    linewidth=1.2
)

# 坐标轴与标题
plt.xlabel(r'$B\ /\ \mathrm{mT}$')
plt.ylabel(r'$U_H\ /\ \mathrm{mV}$')
plt.title(r'$U_H - B$ 曲线')

# 网格
plt.grid(True, linestyle=':', linewidth=0.6)

plt.legend(fontsize=9)
plt.tight_layout()
plt.show()

# ======================== 输出结果 ========================
print(f"拟合结果：U_H = k B + b")
print(f"斜率 k = {k:.5f} mV/mT")
print(f"截距 b = {b:.5f} mV")
print(f"相关系数 r = {r:.6f}")
